python/ray/_private/accelerators/amd_gpu.py

```diff
@@ -43,9 +43,10 @@ def get_current_process_visible_accelerator_ids() -> Optional[List[str]]:
 
         hip_val = os.environ.get(HIP_VISIBLE_DEVICES_ENV_VAR, None)
         if cuda_val := os.environ.get(CUDA_VISIBLE_DEVICES_ENV_VAR, None):
-            assert (
-                hip_val == cuda_val
-            ), f"Inconsistant values found. Please use either {HIP_VISIBLE_DEVICES_ENV_VAR} or {CUDA_VISIBLE_DEVICES_ENV_VAR}."
+            if hip_val is not None and cuda_val is not None:
+                assert (
+                    hip_val == cuda_val
+                ), f"Inconsistant values found. Please use either {HIP_VISIBLE_DEVICES_ENV_VAR} or {CUDA_VISIBLE_DEVICES_ENV_VAR}."
 
         amd_visible_devices = os.environ.get(
             AMDGPUAcceleratorManager.get_visible_accelerator_ids_env_var(), None
```

Comment:
I think we would also need to possibly fetch all the device ids from `CUDA_VISIBLE_DEVICES_ENV_VAR` instead of `HIP_VISIBLE_DEVICES_ENV_VAR` if `CUDA_VISIBLE_DEVICES_ENV_VAR` is only set in this case, since the code here fetches only from `HIP_VISIBLE_DEVICES_ENV_VAR`.

https://github.com/ray-project/ray/blob/8a165da94585d91058a8817c7a607d13348a75b2/python/ray/_private/accelerators/amd_gpu.py#L34-L35